src/lib.rs

```diff
@@ -15,14 +15,8 @@ fn panic() -> ! {
     cortex_m::asm::udf()
 }
 
-#[defmt::timestamp]
-fn timestamp() -> u64 {
-    static COUNT: AtomicUsize = AtomicUsize::new(0);
-    // NOTE(no-CAS) `timestamps` runs with interrupts disabled
-    let n = COUNT.load(Ordering::Relaxed);
-    COUNT.store(n + 1, Ordering::Relaxed);
-    n as u64
-}
+static COUNT: AtomicUsize = AtomicUsize::new(0);
+defmt::timestamp!("{=usize}", COUNT.fetch_add(1, Ordering::Relaxed));
```

Comment:
Note that this won't work on thumbv6, which I do think we should support with the app template. Perhaps move the old load+store sequence to a small function and call that?

Reply:
Will do. Can you quickly explain why it won't work? Or can I read it up somewhere?

Reply:
thumbv6 does not have atomic read-modify-write instructions that are needed to implement `fetch_add`. It can only perform atomic `load` and `store`.

(specifically, thumbv7+ has `ldrex`/`strex` instructions, while thumbv6 doesn't)

Unfortunately the Rust documentation is very bad here and doesn't tell you this.

Reply:
Thanks for the explaination :)
The docs of `AtomicUsize::fetch_add(...)` at least say that:
> **Note:** This method is only available on platforms that support atomic operations on usize.

Then you only need to find out what platforms these are 🙈